lib/ui/painting.dart

```diff
@@ -890,10 +893,12 @@ abstract class Codec extends NativeFieldWrapperClass2 {
   /// * -1 for infinity repetitions.
   int get repetitionCount native "Codec_repetitionCount";
 
-  /// Returns the next animation frame.
+  /// Fetches the next animation frame.
   ///
   /// Wraps back to the first frame after returning the last frame.
-  FrameInfo getNextFrame() native "Codec_getNextFrame";
+  ///
+  /// Returns an error message on failure, null on success.
+  String getNextFrame(NextFrameInfoCallback callback) native "Codec_getNextFrame";
```

Comment:
we'll want to change this to returning void and throwing on error at some point, for consistency with the rest of the API surface.